src/uu/expr/src/syntax_tree.rs

```diff
@@ -110,67 +105,66 @@ impl NumericOp {
                 };
                 a % b
             }
-        }
-        .to_string())
+        }))
     }
 }
 
 impl StringOp {
-    fn eval(&self, left: &AstNode, right: &AstNode) -> ExprResult<String> {
+    fn eval(&self, left: &AstNode, right: &AstNode) -> ExprResult<NumOrStr> {
         match self {
             Self::Or => {
                 let left = left.eval()?;
-                if is_truthy(&left) {
+                if is_truthy(&left.to_string()) {
```

Comment:
I think we can define `is_truthy` on `NumOrStr` now, where `0` of a `BigInt` is false.